vespa/src/main/java/ai/marqo/search/HybridSearcher.java

```
@@ -46,10 +51,47 @@ public class HybridSearcher extends Searcher {
     private static String MARQO_SEARCH_METHOD_TENSOR = "tensor";
     private List<String> STANDARD_SEARCH_TYPES = new ArrayList<>();
 
-    private static class SortField {
-        public String field_name;
-        public String order;
-        public String missing;
+    // Thread-safe ObjectReader for parsing SortField JSON
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final ObjectReader SORT_FIELD_READER =
+            OBJECT_MAPPER.readerFor(new TypeReference<List<SortField>>() {});
+
+    // A magic number used to represent missing sort field values in search results as we can only
+    // return numeric values in match-features.
+    // The value -1e50 is chosen as it is an extremely low number unlikely to occur in real data.
+    private static final double MISSING_SORT_VALUE_SENTINEL = -1e50;
+
+    /**
+     * Represents a field to sort by in the search results.
+     * Uses Java record for immutability and conciseness.
+     */
+    private record SortField(
+            @JsonProperty("field_name") String fieldName,
+            @JsonProperty("order") SortOrder order,
+            @JsonProperty("missing") MissingOrder missing) {}
+
+    /** Sort order enum for better type safety */
+    private enum SortOrder {
+        ASC,
+        DESC;
+
+        @JsonCreator
+        public static SortOrder fromString(String value) {
+            if (value == null) return ASC; // default
+            return "desc".equalsIgnoreCase(value) ? DESC : ASC;
```

A good practice of handling input is `validate input rigorously and fail fast on bad data.` 
So it would be better to let invalid input error out instead of falling back to default. 
`return SortOrder.valueOf(value.toUpperCase(Locale.ROOT));` . This will throw IllegalArgumentException if a value other than `desc` or `asc` is provided.

---

Thanks! Made the change. I decided not to catch the `IllegalArgumentException`.